bin/src/model/pubspec.dart

```diff
@@ -0,0 +1,128 @@
+import 'dart:io';
+
+import 'package:yaml/yaml.dart';
+import '../util/logger.dart';
+import 'dto/dependency.dart';
+import 'pubspec_lock.dart';
+
+const rawGithubDomain = 'https://raw.githubusercontent.com';
+const githubDomain = 'https://github.com';
+const licensePath = '/master/LICENSE';
+const licensePathShort = '/LICENSE';
+
+class Params {
+  static const yamlConfigLicense = 'license_generator';
+  static const yamlConfigLicensesList = 'licenses';
+  static const yamlConfigExtraLicensesList = 'extra_licenses';
+  static const yamlConfigExtraLicenseName = 'name';
+  static const yamlConfigExtraLicenseVersion = 'version';
+  static const yamlConfigExtraLicenseUrl = 'url';
+  static const yamlConfigExtraLicenseLicenseUrl = 'license';
+  static const yamlConfigFailFast = 'failFast';
+  static const yamlConfigCheckBeforeGenerate = 'checkBeforeGenerate';
+  static const yamlConfigDownloadPubDevDetails = 'downloadPubDevDetails';
+
+  final pubspecLock = PubspecLock();
+  String? projectName;
+  var failFast = false;
+  var checkBeforGenerate = false;
+  var downloadPubDevDetails = false;
+  final _dependencies = <Dependency>[];
+  final _dependencyOverrides = <String, String>{};
+
+  Map<String, String> get dependencyOverrides => _dependencyOverrides;
+
+  List<Dependency> get dependencies => _dependencies;
+
+  List<Dependency> get devDependencies => _dependencies.where((element) => element.isDevDependency).toList();
+
+  List<Dependency> get mainDependencies => _dependencies.where((element) => !element.isDevDependency).toList();
+
+  Future<void> init(String pubspecContent, String pubspecLockContent) async {
```

Comment:
There is no reason for this to be async (as the only awaited item is pubspecLock.init which does not need async). This can all move to the constructor